### PR TITLE
[Lua] Enable Third Eye to block ranged attacks

### PR DESCRIPTION
**Pull request by Xaver-DaRed**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Ports https://github.com/AirSkyBoat/AirSkyBoat/pull/3670 from ASB

- As title sais, enables Third Eye to block ranged attacks.

## Steps to test these changes

Find a rng of some type and pick a fight as a sam or /sam
Use third eye and expect to see ranged attacks anticipated

use !exec target:useMobAbility(737) to trigger an EES

### CODE REVIEW

File: scripts/globals/mobskills.lua [hunk 1]
```diff
@@ -557,14 +557,20 @@ xi.mobskills.mobFinalAdjustments = function(dmg, mob, skill, target, attackType,
...
+    then
+        if not skill:isSingle() then
+            target:delStatusEffect(xi.effect.THIRD_EYE)
+        end
 
-    --handle Third Eye using shadowbehav as a guide
-    if attackType == xi.attackType.PHYSICAL and utils.thirdeye(target) then
-        skill:setMsg(xi.msg.basic.ANTICIPATE)
-        return 0
+        --handle Third Eye using shadowbehav as a guide
```
Comment (TeoTwawki) [inline scripts/globals/mobskills.lua]: `--` -> `-- `

surprised a ci rule didn't trip on that

Comment (Xaver-DaRed) [reply scripts/globals/mobskills.lua]: I'm more surprised at me not seing it

Comment (TeoTwawki) [reply scripts/globals/mobskills.lua]: there is another one like it in a section of code you didn't touch. in ability.lua

Comment (Xaver-DaRed) [reply scripts/globals/mobskills.lua]: Fixed all of the ones i could find

**Review by TeoTwawki — APPROVED** [review]

**Review by EpicTaru — APPROVED** [review]

**Pull request merged**